Grab funds tooltip

rin-st opened the pull request:
daisyui doesn't support complex tooltips, like bottom-left, only centered bottom or centered left. So I placed the tooltip to the left (only when the balance is 0)
<img width="305" alt="image" src="https://user-images.githubusercontent.com/25638585/212723223-60f4ee99-a90f-4c39-8493-b249124f4727.png">

bottom variant is bad:
<img width="310" alt="image" src="https://user-images.githubusercontent.com/25638585/212723460-1c64b79e-4503-4fd6-83da-39c366eb6f47.png">


Possible solutions if you want the bottom tooltip:
- move the faucet button to the left
- use something like `react-popper` https://popper.js.org/react-popper/v2/

carletex commented on packages/nextjs/hooks/scaffold-eth/useWalletBalance.ts:
```diff
@@ -0,0 +1,30 @@
+import { useBalance } from "wagmi";
+import { useAppStore } from "~~/services/store/store";
+
+export function useWalletBalance(address?: string) {
```
I like the idea of extracting this functionality as a hook!

The name... what about `useAddressBalance`, or `useAccountBalance`? Seems more appropriate than Wallet.

rin-st replied:
updated to `useAccountBalance`

carletex commented on packages/nextjs/hooks/scaffold-eth/useWalletBalance.ts:
```diff
@@ -0,0 +1,30 @@
+import { useBalance } from "wagmi";
+import { useAppStore } from "~~/services/store/store";
+
+export function useWalletBalance(address?: string) {
```
Why the address arg is optional?

rin-st replied:
Because `wagmi`'s `useAccount` can return an undefined address and `useBalance` is ok with using an undefined address. So it works correct

carletex replied:
I saw that it was working properly, but from the DX perspective is a bit confusing using the hook `useAccountBalance` that has a non-mandatory `address` arg.

But yeah, we need it to be this way to play nicely with wagmi.

Thanks!

rin-st replied:
it's much better than redundant `if-else`'s, problems with conditional hooks, and so on :)

carletex reviewed (changes requested):
Hey @rin-st 

Thanks for this! Happy to see you are interested on collaborating on SE-2 ;)

I left a couple of comments in the code review.

Also, a couple things:

**1. Tooltip always open**

Maybe I wasn't clean enough on the issues (sorry!), but we wan't the tooltip to show all the time (not only on hover) when we are in localhost & balance is 0. You can use the `tooltip-open` class

**2. Tooltip position**
> daisyui doesn't support complex tooltips, like bottom-left, only centered bottom or centered left. So I placed the tooltip to the left (only when the balance is 0)

For this, you can use the daisy `tooltip-bottom` class and add the needed Tailwind classes to look like we want. I managed to do this adding some tw classes (`before:left-auto before:right-0 before:transform-none before:content-[attr(data-tip)]`)

![rinat](https://user-images.githubusercontent.com/2486142/212922151-c660dfb8-e4ac-494a-9c85-66b1c5ece237.png)

You can also make the tooltip tail (the triangle indicator) a bit bigger, overriding the `--tooltip-tail` var. (6px looks good, as in the screenshot)

rin-st commented:
Hey @carletex, thanks for the comments!
1. fixed
2. fixed. Overriding `--tooltip-tail` and `--tooltip-offset` doesn't work for me, so I changed styles by tw classes. It seems there are only a few different vars that can be overridden in daisy-ui. https://daisyui.com/docs/themes/#-5:~:text=There%20are%20a-,few%20optional%20CSS%20variables%20that%20you%20can%20use%20in%20daisyUI%20themes,-to%20customize%20design
<img width="288" alt="image" src="https://user-images.githubusercontent.com/25638585/212962699-e939c1d4-3909-48ad-a930-e5a5b9715506.png">

carletex reviewed (changes requested):
Thanks for the updates! We are almost there:

> 2. fixed. Overriding --tooltip-tail and --tooltip-offset doesn't work for me, so I changed styles by tw classes. It seems there are only a few different vars that can be overridden in daisy-ui. See [#CSS variables in daisyUI themes](https://daisyui.com/docs/themes/#-5)

I don't like the magic number (42) in the TW classes. You can add custom styles to a theme: [Method 1](https://daisyui.com/docs/themes/#-6), [Method 2](https://daisyui.com/docs/themes/#-8).

I prefer Method 2 I think. For both themes.

```js
   //....
  "--rounded-btn": "9999rem",

 // Add this for both themes.
  ".tooltip": {
    "--tooltip-tail": "6px",
  },
```

You only need to add `--tooltip-tail`, since `--tooltip-offset` is calculated based on the the tail.

rin-st commented:
Updated, thanks. Strange, why I can't change the variable globally, but can change it in CSS class?
Checked daisy code - that variables are defined inside `.tooltip` class so it override global vars 🤷‍♂️ .
I don't like magic numbers too (and Tailwind JIT mode), but CSS is CSS 😄 , it's full of magic numbers. Of course we don't need it after `--tooltip-tail` works

carletex reviewed (approved):
Great work @rin-st 

Merging this!
